liquidctl/driver/nzxt_smart_device.py

```diff
@@ -82,33 +119,34 @@
 LOGGER = logging.getLogger(__name__)
 
 _COLOR_MODES = {
-    # (byte2/mode, byte3/variant, byte4/size, min colors, max colors)
-    'off':                           (0x00, 0x00, 0x00, 0, 0),
-    'fixed':                         (0x00, 0x00, 0x00, 1, 1),
-    'super-fixed':                   (0x00, 0x00, 0x00, 1, 40),  # independent leds
-    'fading':                        (0x01, 0x00, 0x00, 1, 8),
-    'spectrum-wave':                 (0x02, 0x00, 0x00, 0, 0),
-    'backwards-spectrum-wave':       (0x02, 0x10, 0x00, 0, 0),
-    'marquee-3':                     (0x03, 0x00, 0x00, 1, 1),
-    'marquee-4':                     (0x03, 0x00, 0x08, 1, 1),
-    'marquee-5':                     (0x03, 0x00, 0x10, 1, 1),
-    'marquee-6':                     (0x03, 0x00, 0x18, 1, 1),
-    'backwards-marquee-3':           (0x03, 0x10, 0x00, 1, 1),
-    'backwards-marquee-4':           (0x03, 0x10, 0x08, 1, 1),
-    'backwards-marquee-5':           (0x03, 0x10, 0x10, 1, 1),
-    'backwards-marquee-6':           (0x03, 0x10, 0x18, 1, 1),
-    'covering-marquee':              (0x04, 0x00, 0x00, 1, 8),
-    'covering-backwards-marquee':    (0x04, 0x10, 0x00, 1, 8),
-    'alternating':                   (0x05, 0x00, 0x00, 2, 2),
-    'moving-alternating':            (0x05, 0x08, 0x00, 2, 2),
-    'backwards-moving-alternating':  (0x05, 0x18, 0x00, 2, 2),
-    'pulse':                         (0x06, 0x00, 0x00, 1, 8),
-    'breathing':                     (0x07, 0x00, 0x00, 1, 8),   # colors for each step
-    'super-breathing':               (0x07, 0x00, 0x00, 1, 40),  # one step, independent leds
-    'candle':                        (0x09, 0x00, 0x00, 1, 1),
-    'wings':                         (0x0c, 0x00, 0x00, 1, 1),
-    'super-wave':                    (0x0d, 0x00, 0x00, 1, 40),  # independent ring leds
-    'backwards-super-wave':          (0x0d, 0x10, 0x00, 1, 40),  # independent ring leds
+    # (byte2/mode, byte3/variant, byte4/size, v2modeflag, min colors, max colors)
+    # v2modeflag is for NZXT SmartDevice V2 only
```

Comment:
It is not strictly necessary to add "v2modeflag" as a new tuple. So the entire field has been reverted back to its original state. As long as mod3 is non-zero, the code issues a 0x01 instead of 0x00.

>  channel_mod = [0x01, 0x20][cid]  # the purpose of this is unknown, but is based on cmd issued by CAM software
>  header = [0x28, 0x03, cid+1, channel_mod, mval, sval, 0x0, [0x00, 0x01][mod3 > 0], color_count, 0x0]